src/Authentication/Authentication.Test/TokenCache/TokenCacheStorageTests.cs

```diff
@@ -0,0 +1,153 @@
+            Assert.Equal(app2StrContent, Encoding.UTF8.GetString(app2StoredBuffer));
+
+            // Cleanup
+            CleanTokenCache(TestAppId1);
+            CleanTokenCache(TestAppId2);
+        }
+
+
+        [Fact]
+        public void ShouldUpdateTokenInPlatformCache()
+        {
+            // Arrange
+            string originalStrContent = "random data for app.";
+            byte[] originalBuffer = Encoding.UTF8.GetBytes(originalStrContent);
+            TokenCacheStorage.SetToken(TestAppId1, originalBuffer);
+
+            // Act
+            string strContentToUpdate = "updated random data for app.";
+            byte[] updateBuffer = Encoding.UTF8.GetBytes(strContentToUpdate);
+            TokenCacheStorage.SetToken(TestAppId1, updateBuffer);
+
+            // Assert
+            byte[] storedBuffer = TokenCacheStorage.GetToken(TestAppId1);
+            Assert.NotEqual(originalBuffer.Length, storedBuffer.Length);
+            Assert.Equal(updateBuffer.Length, storedBuffer.Length);
+            Assert.Equal(strContentToUpdate, Encoding.UTF8.GetString(storedBuffer));
+
+            // Cleanup
+            CleanTokenCache(TestAppId1);
+        }
+
+        [Fact]
+        public void ShouldReturnNoContentWhenPlatformCacheIsEmpty()
+        {
+            // Arrange
+            CleanTokenCache(TestAppId1);
+
+            // Act
+            byte[] storedBuffer = TokenCacheStorage.GetToken(TestAppId1);
+
+            // Assert
+            Assert.Empty(storedBuffer);
+        }
+
+        [Fact]
+        public void ShouldDeleteCache()
+        {
+            // Arrange
+            string originalStrContent = "random data for app.";
+            byte[] originalBuffer = Encoding.UTF8.GetBytes(originalStrContent);
+            TokenCacheStorage.SetToken(TestAppId1, originalBuffer);
+
+            // Act
+            TokenCacheStorage.DeleteToken(TestAppId1);
+
+            // Assert
+            byte[] storedBuffer = TokenCacheStorage.GetToken(TestAppId1);
+            Assert.Empty(storedBuffer);
+        }
+
+
+        [Fact]
+        public void ShouldMakeParallelCallsToTokenCache()
+        {
+            // Arrange
+            int executions = 50;
+            int count = 0;
+            bool failed = false;
+
+            // Act
+            Parallel.For(0, executions, (index) => {
+                byte[] contentBuffer = Encoding.UTF8.GetBytes(index.ToString());
+                TokenCacheStorage.SetToken($"{index}", contentBuffer);
+
+                byte[] storedBuffer = TokenCacheStorage.GetToken(index.ToString());
+                if (index.ToString() != Encoding.UTF8.GetString(storedBuffer))
+                {
+                    failed = true;
+                }
+
+                CleanTokenCache(index.ToString());
+                count++;
+            });
+
+            // Assert
+            Assert.Equal(executions, count);
+            Assert.False(failed);
```

Comment:
consider having a customized message for these failures, it can help parse what these values represent without looking at the code.